File: golang/718_maximum-length-of-repeated-subarray.go
func findLength(A []int, B []int) int {

	lenA := len(A)
	lenB := len(B)
	var r int

	for i := 0; i < lenA && i+r < lenA; i++ {
	LABEL:
		for j := 0; j < lenB && j+r < lenB; j++ {
			k := 0
			for ; i+k < lenA && j+k < lenB; k++ {
				if A[i+k] != B[j+k] {
					if k > r {
						r = k
					}
					continue LABEL
				}

			}
			if k > r {
				r = k
			}

		}

	}

	return r

}

File: java/718_maximum-length-of-repeated-subarray.java
class Solution {
    public int findLength(int[] A, int[] B) {
        int lenB=B.length;
        int r=0;
        int[] emap=new int[lenB+1];
        


                
        for (int v:A){
            for (int i=lenB;i>0;i--){
                if (v==B[i-1]){
                    int lene=emap[i-1]+1;
                    emap[i]=lene;
                    if (lene>r){
                        r=lene;
                    }
                }else{
                    emap[i]=0;
                }
            }
        }
        
        return r;
        
        
    }
}